apps/form-management-app/src/app/state/form/form.slice.ts

```diff
@@ -1,21 +1,234 @@
+import { standardV1JsonSchema, commonV1JsonSchema } from '@abgov/data-exchange-standard';
+import { createDefaultAjv, tryResolveRefs } from '@abgov/jsonforms-components';
+import { JsonFormsCore, JsonSchema, UISchemaElement } from '@jsonforms/core';
 import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
 import axios from 'axios';
+import * as _ from 'lodash';
 import { AppState } from '../store';
-import { FormDefinition, FORM_SERVICE_ID } from '../types';
 import { getAccessToken } from '../user/user.slice';
-
+import { FORM_SERVICE_ID } from '../types';
+import { FormDefinition } from '../types';
 export const FORM_FEATURE_KEY = 'form';
 
+export type ValidationError = JsonFormsCore['errors'][number];
+
 export interface FormState {
   definitions: FormDefinition[];
+  selected: string | null;
   loading: boolean;
+  currentDefinition: FormDefinition | null;
+  editor: {
+    selectedId: string;
+    loading: boolean;
+    saving: boolean;
+    original: FormDefinition;
+    modified: Omit<FormDefinition, 'dataSchema' | 'uiSchema'>;
+    dataSchema: JsonSchema;
+    uiSchema: UISchemaElement;
+    dataSchemaDraft: string;
```

Comment:
we can manage the state for the drafts for UI and data within the container itself, doing it via redux thunk seems like an over kill. local state management should be manged within the conatiner itself, we can store it here only if this data needs to be used by other containers

Reply:
you can remove unused thunks after